redbot/core/core_commands.py

```diff
@@ -2151,47 +2167,68 @@ async def slash_enablecog(self, ctx: commands.Context, cog_name: str):
         names.extend(to_add_message)
         names.extend(to_add_user)
         formatted_names = humanize_list([inline(name) for name in names])
-        await ctx.send(
-            _("Enabled {count} commands from `{cog_name}`:\n{names}").format(
-                count=count, cog_name=cog_name, names=formatted_names
-            )
+        formatted_successful_cogs = humanize_list([inline(name) for name in successful_cogs])
+
+        output = _("Enabled {count} commands from {cog_names}:\n{names}").format(
+            count=count, cog_names=formatted_successful_cogs, names=formatted_names
         )
+        for page in pagify(output):
+            await ctx.send(page)
 
-    @slash.command(name="disablecog")
-    async def slash_disablecog(self, ctx: commands.Context, cog_name):
+    @slash.command(name="disablecog", require_var_positional=True)
+    async def slash_disablecog(self, ctx: commands.Context, *cog_names: str):
         """Marks all application commands in a cog as being disabled, preventing them from being added to the bot.
 
         See a list of cogs with application commands with `[p]slash list`.
 
         This command does NOT sync the enabled commands with Discord, that must be done manually with `[p]slash sync` for commands to appear in users' clients.
 
         **Arguments:**
-            - `<cog_name>` - The cog to disable commands from. This argument is case sensitive.
+            - `<cog_names>` - The cogs to disable commands from. This argument is case sensitive.
         """
         removed = []
+        removed_cogs = set()
         for name, com in self.bot.tree._global_commands.items():
-            if self._is_submodule(cog_name, com.module):
-                await self.bot.disable_app_command(name, discord.AppCommandType.chat_input)
-                removed.append(name)
+            for cog_name in cog_names:
+                if self._is_submodule(cog_name, com.module):
+                    await self.bot.disable_app_command(name, discord.AppCommandType.chat_input)
+                    removed.append(name)
+                    removed_cogs.add(cog_name)
         for key, com in self.bot.tree._context_menus.items():
-            if self._is_submodule(cog_name, com.module):
-                name, guild_id, com_type = key
-                await self.bot.disable_app_command(name, discord.AppCommandType(com_type))
-                removed.append(name)
+            for cog_name in cog_names:
+                if self._is_submodule(cog_name, com.module):
+                    name, guild_id, com_type = key
+                    await self.bot.disable_app_command(name, discord.AppCommandType(com_type))
+                    removed.append(name)
+                    removed_cogs.add(cog_name)
         if not removed:
```

Comment:
Since this one has nothing that would cause it to not be able to disable any particular command, there's no worries about this one being atomic. Still, this command currently does not properly display when an *individual* cog did not have any commands, only when *all* cogs did not have commands, due to this line. If you changed this to something like this, that would be sufficient for this particular command:
```py
        failed_cogs = set(cog_names) - removed_cogs
        if failed_cogs:
            # use failed_cogs instead of cog_names for processing & user output
            ...
        if not removed_cogs:
            return
```